### ISSUE TITLE
使用了MDConv的MDDA_former网络结构的请求

### ISSUE THREAD

**Issue by 777sherlock** [House-yuyu/MDDA-former#3]
作者，您好。 我仔细看了您的文章，发现你在文中着重强调了MDConv的重要性，但是我在您目前提供的代码里面还没找到相关代码，MDDA_former.py里面用的好像是ODConv，请问作者能更新一下代码吗，用作学习，感谢

**Comment by House-yuyu** [comment]
代码中ODConv的名称没改（实际上应该是MDConv），和ODConv的区别是我们舍弃了其中一个维度来最求性能和复杂度的平衡（我们没有用参数量增加最大的那个kernel维度，具体可以比较下论文中图4和ODConv原文中的区别）

**Comment by 777sherlock** [comment]
好的，感谢解答

**Comment by 777sherlock** [comment]
你好，我又仔细看了一下您提供的代码，发现MDDA_former.py里面用的好像就是ODConv（具有通道注意力、滤波器注意力、空间注意力 和 卷积核注意力），而不是（通道注意力、滤波器注意力、空间注意力），但是论文里面确实是说MDConv只用了（通道注意力、滤波器注意力、空间注意力）。但是跟代码里面的好像对不上，能解答一下吗？

**Comment by 777sherlock** [comment]
channel_attention, filter_attention, spatial_attention, kernel_attention = self.attention(x)
x = x * channel_attention
aggregate_weight = spatial_attention * kernel_attention * self.weight.unsqueeze(dim=0)
aggregate_weight = torch.sum(aggregate_weight, dim=1).view(
    [-1, self.in_planes // self.groups, self.kernel_size, self.kernel_size])
output = F.conv2d(x, weight=aggregate_weight, bias=None, stride=self.stride, padding=self.padding,
                  dilation=self.dilation, groups=self.groups * batch_size)
output = output * filter_attention
这是代码里面所使用到的注意力，我有点愚钝，为什么只使用了3种呢

**Comment by House-yuyu** [comment]
@777sherlock , hi, 和kernel_attention和kernel_num有关系，num为1时就不存在kernel_attention了, 我们在网络里定义的数量为1
具体可见
utils/odconv.py#L35-L39

**Comment by House-yuyu** [comment]
@777sherlock Hi, kernel_attention和kernel_num有关系，num为1时就不存在kernel_attention了, 我们在网络里定义的数量为1
具体可见
MDDA-former/utils/odconv.py#L35-L39

**Comment by House-yuyu** [comment]
utils/odconv.py#L35-L39

**Comment by House-yuyu** [comment]
> utils/odconv.py#L35-L39

**Comment by 777sherlock** [comment]
哦哦 大概明白了 感谢
